z3c/dependencychecker/dotted_name.py

```diff
@@ -29,19 +29,18 @@ def from_requirement(cls, requirement, file_path=None):
             file_path=file_path,
         )
 
-    @staticmethod
-    def _get_safe_name(name):
-        safe_name = name.lower().replace('-', '_')
-        return safe_name
+    @cached_property
+    def safe_name(self):
+        self._safe_name = self._safe_name.lower().replace('-', '_')
```

Comment:
Setting `self._safe_name` is a bit of side effect (and thus dangerous, as it is only set after the first time you've called `self.safe_name`.

In this case I'd probably put `self.safe_name = .....lower().replace()` directly in the class's init (or keep the original method call, but with a different name).

Reply:
Ok, I moved it to the ``__init__``